File: frontend/src/interfaces/hospital.ts
/**
 * @module hospital
 * @file hospital.ts
 * @author eunchang
 * @date 2025-03-26
 * @description 병원 도메인 전용 인터페이스 모듈입니다.
 *
 * 동물병원 서비스 구현시 필요한 인터페이스들을 정의합니다.
 *
 * ===========================================================
 * DATE              AUTHOR             NOTE
 * -----------------------------------------------------------
 * 2025-03-26        eunchang         최초 생성
 */

/**
 * Hospital Base Info interface
 */
export interface HospitalBase {
  id: number;
  name?: string;
  address?: string;
  phone_number?: string;
  authorization_number?: string;
  x_axis?: number;
  y_axis?: number;
}

/**
 * Hospital Detail Info Interface
 */
export interface HospitalDetail extends HospitalBase {
  phone_number?: string;
  did?: string;
  account?: string;
  password?: string;
  public_key?: string;
  delete_date?: string;
  email?: string;
  doctor_name?: string;
}

File: backend/KKUKKKUK/src/main/java/com/be/KKUKKKUK/domain/hospital/entity/Hospital.java
package com.be.KKUKKKUK.domain.hospital.entity;

import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.math.BigDecimal;
import java.time.LocalDate;

/**
 * packageName    : com.be.KKUKKKUK.domain.hospital.entity<br>
 * fileName       : Hospital.java<br>
 * author         : Fiat_lux<br>
 * date           : 2025-03-13<br>
 * description    : hospital entity 클래스입니다. <br>
 * ===========================================================<br>
 * DATE              AUTHOR             NOTE<br>
 * -----------------------------------------------------------<br>
 * 25.03.13          Fiat_lux           최초생성<br>
 * 25.03.14          haelim             licenseNumber, doctorName 필드 추가
 * 25.03.14          haelim             licenseNumber, 필드 삭제
 * 25.04.01          haelim             publicKey, 필드 삭제
 */

@Entity
@Table(name = "hospital")
@Getter
@Setter
@Builder
@ToString
@AllArgsConstructor
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@EntityListeners(AuditingEntityListener.class)
public class Hospital {
    @Id
    @Column(nullable = false)
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(length = 100, nullable = false)
    private String name;

    @Column(name = "phone_number", length = 14)
    private String phoneNumber;

    @Column
    private String did;

    @Column(length = 100)
    private String account;

    @Column(length = 100)
    private String password;

    @Column(nullable = false)
    private String address;

    @Column(name = "flag_certified", nullable = false)
    private Boolean flagCertified;

    @Column(name = "delete_date")
    private LocalDate deleteDate;

    @Column(name = "authoriazation_number", nullable = false)
    private String authorizationNumber;

    @Column(name = "x_axis", nullable = false)
    private BigDecimal xAxis;

    @Column(name = "y_axis", nullable = false)
    private BigDecimal yAxis;

    @Column(name = "doctor_name")
    private String doctorName;

    @Column(name = "email")
    private String email;

}
